presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestMaterializedViewQueryOptimizer.java

```diff
@@ -388,13 +388,13 @@ public void testWithLimitClause()
     @Test
     public void testWithUnsupportedFunction()
     {
-        String originalViewSql = format("SELECT AVG(a) FROM %s", BASE_TABLE_1);
```

Comment:
This case should still not work? Our implementation relies on the fact that AVG queries relies on the view query having SUM and count functions. Which is not the case here.

Reply:
This test will still pass, but I felt it was prudent to remove `avg` from our definition of "unsupported functions" for the test. I added tests that try to rewrite `avg` and `approx_distinct` without the necessary derived columns in view table.

Reply:
The reason I wanted to keep this, as our support is still very limited. Many of the practical usecases might still not work.